src/Aspire.Hosting/Utils/HashDigestStream.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO.Hashing;
+
+namespace Aspire.Hosting.Utils;
+
+/// <summary>
+/// A stream capable of computing the hash digest of raw data while also copying it.
+/// </summary>
+sealed class HashDigestStream : Stream
+{
+    private readonly Stream _writeStream;
+    private readonly NonCryptographicHashAlgorithm _hashAlgorithm;
+
+    public HashDigestStream(Stream writeStream, NonCryptographicHashAlgorithm hashAlgorithm)
+    {
+        _writeStream = writeStream;
+        _hashAlgorithm = hashAlgorithm;
+    }
+
+    public override bool CanWrite => true;
+
+    public override void Write(byte[] buffer, int offset, int count)
+    {
+        _hashAlgorithm.Append(buffer.AsSpan(offset, count));
+        _writeStream.Write(buffer, offset, count);
+    }
+
+    public override void Write(ReadOnlySpan<byte> buffer)
+    {
+        _hashAlgorithm.Append(buffer);
```

Comment:
nit, I think this method is dead code (?)

Reply:
Confirmed, and safe-guarded it by throwing an exception